src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,53 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addToAlive(animalObj) {
+    this.alive.push(animalObj);
+  }
+
+  static removeFromAlive(animalObj) {
+    const animalIndex = this.alive.indexOf(animalObj);
+
+    this.alive.splice(animalIndex, 1);
+  }
+
+  constructor(name, health) {
+    const defaultHealth = 100;
+
+    this.name = name;
+    this.health = health || defaultHealth;
+
+    Animal.addToAlive(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    const bitePower = 50;
+    const isHerbivore = victim instanceof Herbivore;
+
+    if (!victim.hidden && isHerbivore) {
```

Comment:
better to switch the places. If Carnivore is passed you have redundant check for `hidden`